person.rb

```diff
@@ -13,9 +15,43 @@ def of_age?
     (@age.is_a? Numeric) && (@age >= 18)
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def can_use_services?
     @age >= 18 || @parent_permission
   end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```

Comment:
- Hey, :wave: in here, you want to capitalize the first letter and not the whole text as the upcase would make the text all in uppercase. What you want to use is the capitalize method to achieve that. let's update this :pray:

Reply:
Thanks for the information, I will change it to `capitalize` method

Reply:
Thanks for the information, I will change it to `capitalize` method